src/test/utils/ciTelemetry/ciTelemetryReporter.node.ts

```diff
@@ -0,0 +1,32 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import * as os from 'os';
+import { BaseTelemetryReporter } from './baseCiTelemetryReporter';
+import { TelemetryAppender } from './TelemetryAppender';
+import { appInsightsClientFactory } from './appInsightsClientFactory.node';
+
+/**
+ * An isolated wrapper for an Application Insights client that is specifically for sending telemetry during CI jobs.
+ * This won't run on a users machine, so there is no need to check for opt-in status.
+ */
+export class CiTelemetryReporter extends BaseTelemetryReporter {
```

Comment:
Instead of pulling in copies of the code vscode-extension-telemetry, could you just extend the TelemetryReporter in the extension-telemetry npm module?

Reply:
not the way it's currently written - getTelemetryLevel is a utility function which can't be overridden, and is used as a guard on multiple levels

Reply:
Not sure I'm following. Sorry. 

getTelemetryLevel isn't overridden here? 

Did you change it in the BaseTelemetryReporter?

Reply:
I removed getTelemetryLevel entirely since this will only collect telemetry in CI, but in the npm package, it's a global utility function that checks the vscode setting.

Reply:
Could you monkey patch it in the npm code? Just remove it at run time?

I guess I'm not too sure about making a copy of the extension-telemetry code (with modifications).

Reply:
Monkey patching in the test only code sounds safer to me than copying the code over.